{{cookiecutter.project_name}}/.pre-commit-config.yaml

```diff
@@ -0,0 +1,79 @@
+default_stages: [commit]
+repos:
+{%- if cookiecutter.hook_isort == "y" %}
+  - repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+        name: isort (python)
+      - id: isort
+        name: isort (cython)
+        types: [cython]
+      - id: isort
+        name: isort (pyi)
+        types: [pyi]
+{%- endif %}
+{%- if cookiecutter.hook_black == "y" %}
+  - repo: https://github.com/psf/black
+    rev: 23.1.0
+    hooks:
+      - id: black
+{%- endif %}
+{%- if cookiecutter.hook_flake8 == "y" %}
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+      - id: flake8
+        additional_dependencies: [dlint]
+{%- endif %}
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-case-conflict
+      - id: check-json
+      - id: check-merge-conflict
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+{%- if cookiecutter.hook_eslint == "y" %}
+  - repo: https://github.com/eslint/eslint
+    rev: v8.35.0
+    hooks:
+      - id: eslint
+        args: ["monkey/monkey_island/cc/ui/src/", "--fix", "--max-warnings=0"]
+{%- endif %} 
+{%- if cookiecutter.hook_mypy == "y" %}
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.0.1
+    hooks:
+      - id: mypy
+        additional_dependencies: [types-ipaddress, types-paramiko, types-python-dateutil, types-pytz, types-PyYAML, types-requests]
+        exclude: "vulture_allowlist.py"
+        args: [--ignore-missing-imports, --check-untyped-defs] 
+{%- endif %}  
+{%- if cookiecutter.hook_shellcheck == "y" %}  
+  - repo: https://github.com/koalaman/shellcheck-precommit
+    rev: v0.9.0
+    hooks:
+    -   id: shellcheck
+  - repo: local
+    hooks:
+      - id: pytest
+        name: pytest
+        entry: bash -c "cd monkey && pytest"
+        language: system
+        files: "monkey/"
+        exclude: "monkey/monkey_island/cc/ui"
```

Comment:
We want to avoid leaking anything monkey-specific into any of these files This isn't the the cookiecutter template for https://github.com/guardicore/monkey, it's the template the monkey team will use to create a variety of different repositories relating to Infection Monkey.

Most of my comments so far have been about this. See:

- https://github.com/guardicode/infection-monkey-cookiecutter/pull/1#pullrequestreview-1598218373
- https://github.com/guardicode/infection-monkey-cookiecutter/pull/1#discussion_r1313012360
- https://github.com/guardicode/infection-monkey-cookiecutter/pull/1#discussion_r1309045718
- https://github.com/guardicode/infection-monkey-cookiecutter/pull/1#discussion_r1309043878

Reply:
should I make these fields empty, these options would be configurable after the template creation and too many options in cookiecutter.json might make the process of filling out the details much longer. This might be one of the suggestions rest I can create an option if you want.

Reply:
Yeah, empty is fine.